crates/adapters/okx/src/http/client.rs

```diff
@@ -1334,4 +1719,845 @@ mod tests {
         let symbols = client.get_cached_symbols();
         assert_eq!(symbols, vec![first.inst_id.to_string()]);
     }
+
+    // Test endpoint selection logic as per Section 2
+    #[rstest]
```

Comment:
Tests are much better, but lets remove any which don't directly hit the client logic.